local-tests/test_catch_up.py

```diff
@@ -33,45 +36,45 @@
 
 chain.set_flags_validator('validator')
 
-print('Starting the chain')
+printt('Starting the chain')
 chain.start('aleph')
 
-print('Waiting 90s')
-sleep(90)
-
-finalized_before_kill_per_node = check_finalized(chain)
-print('Killing one validator and one nonvalidator')
+printt('Waiting for finalization')
+chain.wait_for_finalization(0)
+printt('Waiting for authorities')
+chain.wait_for_authorities()
 
-chain[3].stop()
-chain[4].stop()
+printt('Killing one validator and one nonvalidator')
+chain.stop(nodes=[3, 4])
+finalized_before_kill = check_finalized(chain)
 
-print('waiting around 2 sessions')
+printt('Waiting around 2 sessions')
 sleep(30 * 2)
 
-print('restarting nodes')
-finalized_before_start_per_node = check_finalized(chain)
+finalized_before_start = check_finalized(chain)
 
 # Check if the finalization didn't stop after a kill.
-if finalized_before_start_per_node[0] - finalized_before_kill_per_node[0] < 10:
-    print('Finalization stalled')
+if finalized_before_start[0] - finalized_before_kill[0] < 10:
```

Comment:
nit: I know that it is sufficient, but I think that `max(finalized_before_start) - max(finalized_before_kill)` is a tad better

Reply:
With `max` we can compare two numbers from two different nodes. How is this better?

Reply:
`max` stands for consensus state in general; possible difference between `max` and single node state should be only due to communication layer

Reply:
Please mind the fact that this information is taken from the logfile, where the corresponding line is put once every 5 seconds. So it's not only the communication layer, differences mostly come from the fact that the general "heartbeat" of the whole thing being these long 5 seconds and nodes obviously don't synchronize when to put that line into logs.